datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -244,43 +251,45 @@ protected void doExecute(SdxContext context, SdxEvent payload,
             @Override
             protected Object getFailurePayload(SdxEvent payload, Optional<SdxContext> flowContext,
                     Exception e) {
-                LOGGER.error("Attach new cluster state Detach action failed.", e);
-                return SdxDetachFailedEvent.from(payload, e);
+                LOGGER.error("Failed to attach new cluster during detach.", e);
+                return new SdxStartDetachRecoveryEvent(payload.getResourceId(), payload.getUserId(), e);
             }
         };
     }
 
-    @Bean(name = "SDX_ATTACH_NEW_CLUSTER_FAILED_STATE")
-    public Action<?, ?> sdxAttachNewClusterFailedAction() {
-        return new AbstractSdxAction<>(SdxDetachFailedEvent.class) {
+    @Bean(name = "SDX_DETACH_RECOVERY_STATE")
+    public Action<?, ?> sdxDetachRecoveryAction() {
+        return new AbstractSdxAction<>(SdxStartDetachRecoveryEvent.class) {
             @Override
             protected SdxContext createFlowContext(FlowParameters flowParameters,
                     StateContext<FlowState, FlowEvent> stateContext,
-                    SdxDetachFailedEvent payload) {
+                    SdxStartDetachRecoveryEvent payload) {
                 return SdxContext.from(flowParameters, payload);
             }
 
             @Override
-            protected void doExecute(SdxContext context, SdxDetachFailedEvent payload,
+            protected void doExecute(SdxContext context, SdxStartDetachRecoveryEvent payload,
                     Map<Object, Object> variables) throws Exception {
-                SdxCluster resized = (SdxCluster) variables.get(RESIZED_SDX);
-                SdxCluster clusterToReattach = (SdxCluster) variables.get(DETACHED_SDX);
-                LOGGER.error("Failed to attach new cluster with ID {} during detaching of cluster with ID {}." +
-                        " Attempting to reattach detached cluster.", resized.getId(), clusterToReattach.getId());
-
+                SdxCluster clusterToReattach = sdxService.getById(payload.getResourceId());
                 clusterToReattach = sdxAttachService.reattachDetachedSdxCluster(clusterToReattach);
-
-                LOGGER.info("Successfully restored detached SDX with ID {} which failed during attaching of new cluster.",
-                        clusterToReattach.getId());
-                sendEvent(context, SDX_DETACH_FAILED_EVENT.event(), payload);
+                LOGGER.info("Successfully restored detached SDX with ID {}.", clusterToReattach.getId());
+
+                // Transition to next step according to if automatic recovery or manual.
```

Comment:
remove the comment and refactor the code so it would be obvious without the comment
eg creating methods with the right naming

Reply:
I removed the comment, I think the code is clear enough.